invisible_cities/reco/deconv_functions_test.py

```diff
@@ -0,0 +1,174 @@
+    drop_function = drop_isolated_sensors(dist, ['E'])
+    df_cut        = drop_function(df)
+
+    if len(df_cut > 0):
+        assert np.isclose(df_cut.E.sum(), df.E.sum())
+
+    for row in df_cut.itertuples(index=False):
+        n_neighbours = len(df_cut[in_range(df_cut.X, row.X - dist[0], row.X + dist[0] + 1e-8) & in_range(df_cut.Y, row.Y - dist[1], row.Y + dist[1] + 1e-8)])
+        assert n_neighbours > 1
+
+def test_interpolateSignal():
+    ref_interpolation = np.array([0.   , 0.   , 0.   , 0.   , 0.   , 0.   , 0.   , 0.   , 0.   ,
+                                  0.   , 0.   , 0.   , 0.   , 0.17 , 0.183, 0.188, 0.195, 0.202,
+                                  0.2  , 0.2  , 0.19 , 0.181, 0.169, 0.   , 0.   , 0.308, 0.328,
+                                  0.344, 0.353, 0.363, 0.365, 0.356, 0.345, 0.327, 0.305, 0.   ,
+                                  0.   , 0.501, 0.531, 0.569, 0.583, 0.596, 0.598, 0.583, 0.566,
+                                  0.533, 0.496, 0.   , 0.   , 0.693, 0.752, 0.784, 0.819, 0.836,
+                                  0.833, 0.82 , 0.794, 0.752, 0.688, 0.   , 0.   , 0.813, 0.882,
+                                  0.922, 0.958, 0.976, 0.975, 0.958, 0.927, 0.88 , 0.813, 0.   ,
+                                  0.   , 0.812, 0.876, 0.929, 0.958, 0.974, 0.975, 0.959, 0.923,
+                                  0.88 , 0.822, 0.   , 0.   , 0.688, 0.752, 0.8  , 0.822, 0.831,
+                                  0.833, 0.819, 0.789, 0.753, 0.693, 0.   , 0.   , 0.496, 0.535,
+                                  0.567, 0.587, 0.597, 0.591, 0.581, 0.57 , 0.532, 0.504, 0.   ,
+                                  0.   , 0.305, 0.326, 0.346, 0.356, 0.362, 0.363, 0.356, 0.342,
+                                  0.33 , 0.31 , 0.   , 0.   , 0.168, 0.18 , 0.189, 0.198, 0.202,
+                                  0.199, 0.195, 0.192, 0.181, 0.174, 0.   , 0.   , 0.   , 0.   ,
+                                  0.   , 0.   , 0.   , 0.   , 0.   , 0.   , 0.   , 0.   , 0.   ])
+
+    g = multivariate_normal((0.5, 0.5), (0.05, 0.5))
+
+    grid_x, grid_y = np.mgrid[0:1:12j, 0:1:12j] #Grid for interpolation
+    points = np.meshgrid(np.linspace(0, 1, 6), np.linspace(0, 1, 6)) # Coordinates where g is known
+    points = (points[0].flatten(), points[1].flatten())
+    values = g.pdf(list(zip(points[0], points[1]))) # Value of g at the known coordinates.
+    n_interpolation = 12 #How many points to interpolate
+
+    out_inter = interpolateSignal(values, points, (np.linspace(-0.05, 1.05, 6), np.linspace(-0.05, 1.05, 6)), [n_interpolation, n_interpolation], 'cubic')
+
+    assert np.allclose(ref_interpolation, np.around(out_inter[0].flatten(), decimals=3))
+    assert np.allclose(shift_to_bin_centers(np.linspace(-0.05, 1.05, n_interpolation + 1)), sorted(set(out_inter[1][0])))
+    assert np.allclose(shift_to_bin_centers(np.linspace(-0.05, 1.05, n_interpolation + 1)), sorted(set(out_inter[1][1])))
```

Comment:
give meaningful name to the `out_inter` items and to the `np.linspace` outcomes for a better readability of the test